classes/Project/Project.mjs

```diff
@@ -140,6 +141,63 @@ export default class Project {
     }
   }
 
+  async addLayer(layer) {
+    const label = asLanguageMap(layer.label ?? `${this.data.label} Layer ${this.data.layers.length + 1}`)
+    const canvases = layer.canvases
+
+    try {
+      const responses = await Promise.all(canvases.map(canvas => fetch(canvas)))
+      const data = await Promise.all(responses.map(response => response.json()))
+      const layerAnnotationCollection = {
+        "@context": "http://www.w3.org/ns/anno.jsonld",
+        "id": `${process.env.RERUMIDPREFIX}${database.reserveId()}`,
+        type: "AnnotationCollection",
+        label,
+        pages: await Promise.all(data.map(async (canvas) => {
```

Comment:
The things of type "AnnotationCollection" does not support a `pages` property.  A type "AnnotationCollection" has the `items` property for this.  

Note that a private TPEN Mongodb Layer has `pages`.